File: IT/Containers/Esercizi/Java/Esercizio29.java
/* file Esercizio29.java */

import java.util.Map;
import java.util.NavigableMap;
import java.util.TreeMap;

public class Esercizio29 {
	public static void main(String [] args) {
		// mappa ordinata che posiziona le coppie chiave/valore in base alla chiave
		NavigableMap<Integer, String> colors = new TreeMap<Integer, String>();

		colors.put(1, "rosso");
		colors.put(2, "verde");
		colors.put(3, "blu");
		colors.put(4, "giallo");
		colors.put(5, "ciano");
		colors.put(6, "magenta");
		colors.put(7, "arancio");
		colors.put(8, "viola");
		colors.put(9, "bianco");
		colors.put(10, "nero");

		System.out.println("Colori = " + colors);

		int k = 1;
		Map<Integer, String> rgb = colors.subMap(k, k += 3);
		Map<Integer, String> cmy = colors.subMap(k, k += 3);
		Map<Integer, String> tc = colors.subMap(k, k += 2);
		Map<Integer, String> nc = colors.subMap(k, k += 2);

		System.out.println("Colori primari = " + rgb);
		System.out.println("Colori secondari = " + cmy);
		System.out.println("Colori terziari = " + tc);
		System.out.println("Colori neutri = " + nc);
	}
}

File: IT/1/Containers/Esercizi/C++/esercizio29.cpp
/*
file esercizio29.cpp
compilare con: g++ -std=c++11 esercizio29.cpp
*/

#include <iostream>
#include <map>

using std::cout;
using std::endl;
using std::string;
using std::map;

int main()
{
	map<int, string> colors = {{1, "rosso"}, {2, "verde"}, {3, "blu"}, {4, "giallo"}, {5, "ciano"}, {6, "magenta"}, {7, "arancio"}, {8, "viola"}, {9, "bianco"}, {10, "nero"}};

	cout << "Colori = {";

	for (auto& col : colors) {
		cout << col.first << "=" << col.second << ", ";
	}

	cout << "\b\b}" << endl;

	map<int, string> rgb(colors.find(1), colors.find(4));
	map<int, string> cmy(colors.find(4), colors.find(7));
	map<int, string> tc(colors.find(7), colors.find(9));
	map<int, string> nt(colors.find(9), colors.end());

	cout << "Colori primari = {";

	for (auto& col : rgb) {
		cout << col.first << "=" << col.second << ", ";
	}

	cout << "\b\b}" << endl;

	cout << "Colori secondari = {";

	for (auto& col : cmy) {
		cout << col.first << "=" << col.second << ", ";
	}

	cout << "\b\b}" << endl;

	cout << "Colori terziari = {";

	for (auto& col : tc) {
		cout << col.first << "=" << col.second << ", ";
	}

	cout << "\b\b}" << endl;

	cout << "Colori neutri = {";

	for (auto& col : nt) {
		cout << col.first << "=" << col.second << ", ";
	}

	cout << "\b\b}" << endl;
}

